src/components/Explore/Explore.tsx

```diff
@@ -267,9 +291,13 @@ const ExploreCopy: React.FunctionComponent<{
           (fips: string) => regions.findByFipsCode(fips)!,
         );
         setSelectedLocations(locations);
-        setMetricMenuLabel(metricLabels[sharedParams.currentMetric]);
+        setRegionNamesMenuLabel(locations.map(getLocationLabel).join('; '));
       }
-    }, [sharedParams, metricLabels]);
+    }, [setCurrentMetric, sharedParams]);
+
+    useEffect(() => {
+      setMetricMenuLabel(metricLabels[currentMetric]);
```

Comment:
moving this setState into its own effect (and out of the one above) seems to have fixed the broken share links